admin/CopyReplication

```diff
@@ -12,35 +12,114 @@ cd "$MB_SERVER_ROOT"
 
 source admin/config.sh
 
-TEMP_DIR=$1 ; shift
+TEMP_DIR="$1" ; shift
 
-# Was a replication packet created?
+cd "$TEMP_DIR"
+
+set_backup_permissions() {
+    chown "$BACKUP_USER:$BACKUP_GROUP" "$1"
+    chmod "$2" "$1"
+}
+
+set_ftp_permissions() {
+    chown "$FTP_USER:$FTP_GROUP" "$1"
+    chmod "$2" "$1"
+}
+
+create_backup_subdir() {
+    local subdir="$BACKUP_DIR"/"$1"
+    if [ ! -d "$subdir" ]
+    then
+        mkdir -p "$subdir"
+        set_backup_permissions "$subdir" "$BACKUP_DIR_MODE"
+    fi
+}
+
+create_ftp_subdir() {
+    local subdir="$FTP_DATA_DIR"/"$1"
+    if [ ! -d "$subdir" ]
+    then
+        mkdir -p "$subdir"
+        set_ftp_permissions "$subdir" "$FTP_DIR_MODE"
+    fi
+}
+
+declare -a new_packets
 shopt -s nullglob
-REPLICATION_FILE=`echo -n "$TEMP_DIR"/replication-*.tar.bz2`
-REPLICATION_FILE_ASC=`echo -n "$TEMP_DIR"/replication-*.tar.bz2.asc`
+for packet in replication-*.tar.bz2
+do
+    new_packets+=("$packet")
+done
```

Comment:
More simple and still correctly handling potential filenames containing whitespace:
```suggestion
declare -a new_packets=(replication-*.tar.bz2)
```

Reply:
Thanks for the bash tips!